core-web/libs/sdk/angular/src/lib/components/dot-editable-text/dot-editable-text.component.ts

```diff
@@ -0,0 +1,266 @@
+import { EditorComponent, TINYMCE_SCRIPT_SRC } from '@tinymce/tinymce-angular';
+import { EventObj } from '@tinymce/tinymce-angular/editor/Events';
+
+import {
+    Component,
+    ElementRef,
+    HostListener,
+    inject,
+    Input,
+    OnChanges,
+    OnInit,
+    Renderer2,
+    SecurityContext,
+    ViewChild
+} from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+
+import {
+    CUSTOMER_ACTIONS,
+    DotCmsClient,
+    isInsideEditor,
+    postMessageToEditor
+} from '@dotcms/client';
+
+import { TINYMCE_CONFIG, TINYMCE_FORMAT, TINYMCE_MODE } from './utils';
+
+import { DotCMSContentlet } from '../../models';
+
+/**
+ * Dot editable text component.
+ * This component is responsible to render a text field that can be edited inline.
+ *
+ * @export
+ * @class DotEditableTextComponent
+ * @implements {OnInit}
+ * @implements {OnChanges}
+ */
+@Component({
+    selector: 'dot-editable-text',
+    standalone: true,
+    templateUrl: './dot-editable-text.component.html',
+    styleUrl: './dot-editable-text.component.css',
+    imports: [EditorComponent],
+    providers: [
+        {
+            provide: TINYMCE_SCRIPT_SRC,
+            useFactory: () => {
+                return `${DotCmsClient.dotcmsUrl}/html/js/tinymcev7/tinymce.min.js`;
+            }
+        }
+    ]
+})
+export class DotEditableTextComponent implements OnInit, OnChanges {
+    @ViewChild(EditorComponent) editorComponent!: EditorComponent;
+
+    /**
+     * Represents the mode of the editor which can be `plain`, `minimal`, or `full`
+     *
+     * @type {TINYMCE_MODE}
+     * @memberof DotEditableTextComponent
+     */
+    @Input() mode: TINYMCE_MODE = 'plain';
```

Comment:
@rjvelazco Why not use signal inputs?

Reply:
@nicobytes There is a `Typescript` error in `Signal Inputs` between Angular versions. Our SDK uses `@angular/core` v17.0.1, this error is shown when you use the component with `@angular/core` v18.0.1

![type-error-signal-inputs](https://github.com/user-attachments/assets/7bc8d41c-14fc-416c-99a7-34db713b3cd2)

Reply:
We can use Signal Inputs to replace these inputs once they become stable. Even though the API doesn't change, the type can vary between versions. This can lead to errors when developing components intended for use in different Angular versions.

Reply:
And also... @KevinDavilaDotCMS is looking to make our SDK to work with Angular 15+ because let's be honest... how many Angular projects are actually in 18?